### ISSUE TITLE
Builds Fail with 1.5.0 of subosito/gotenv

### ISSUE THREAD

**Issue by arhea** [spf13/viper#1600]
### Preflight Checklist

- [X] I have searched the [issue tracker](https://www.github.com/spf13/viper/issues) for an issue that matches the one I want to file, without success.
- [X] I am not looking for support or already pursued the available [support channels](https://github.com/spf13/viper/issues/new/choose) without success.
- [X] I have checked the [troubleshooting guide](https://github.com/spf13/viper/blob/master/TROUBLESHOOTING.md) for my problem, without success.

### Viper Version

1.16.0

### Go Version

1.20

### Config Source

Environment variables

### Format

Dotenv

### Expected Behavior

Package to build.

### Actual Behavior

```
/go/pkg/mod/github.com/spf13/viper@v1.16.0/internal/encoding/dotenv/codec.go:51:33: cannot use &buf (value of type *bytes.Buffer) as gotenv.Reader value in argument to gotenv.StrictParse: *bytes.Buffer does not implement gotenv.Reader (missing method ReadAt)
failed to build, error: exit status 1
```

### Steps To Reproduce

1. Upgrade dependencies of project resulting in `subosito/gotenv` v1.5.0
2. Error message listed above.
3. Adding `replace github.com/subosito/gotenv => github.com/subosito/gotenv v1.4.2` to `go.mod`, builds succeed.

### Additional Information

https://github.com/subosito/gotenv/issues/27

**Comment by guybarnhartmagen** [comment]
this is a duplicate of https://github.com/spf13/viper/issues/1601

**Comment by andig** [comment]
That is actually closed. Is it possible that this is a consequence of not requiring a Go version that has `bytes.Buffer` implement `ReadAt`?

**Comment by arhea** [comment]
It looks like the issue I opened on https://github.com/subosito/gotenv/issues/27 has been resolved, thank you @subosito! As for #1601 it looks like that was created shortly after this issue (#1600), apologies for the duplicates. 

I have confirmed with 1.6.0 that is working again.

I think it is safe to close this issue.
